pkg/providers/cloudstack/cloudstack.go

```diff
@@ -367,6 +419,30 @@ func (p *cloudstackProvider) validateEnv(ctx context.Context) error {
 	return nil
 }
 
+func (p *cloudstackProvider) validateSecretsUnchanged(ctx context.Context, cluster *types.Cluster) error {
```

Comment:
The current validator does not have access to kubectl client. It looks like it is responsible for validations via cmk. 
```
type Validator struct {
	cmk ProviderCmkClient
}
```

Reply:
I feel like the validator should support all cloudstack provider validations, so I'd be inclined to add this functionality there and keep the cloudstack.go provider class as simple as we can

Reply:
Moving the function to validator requires some level of refactoring because the current validator does not have access to kubectl client. I will add a TODO comment and let's revisit this refactoring later.